### PR TITLE
Update cedar-policy and related crates to 3.0.0

### PR DESCRIPTION
**Pull request by adamrothman**
## Description of changes

Update cedar-policy and related crates to 3.0.0. This was mostly just adding `Option<&Schema>` arguments to places where `Request::new()` is called.

## Issue #, if available

N/A

## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [x] A change "invisible" to users (e.g., documentation, changes to "internal" crates, etc.)

I confirm that this PR (choose one, and delete the other options):

- [x] Does not update the CHANGELOG because my change does not significantly impact released code.

## Testing

Ran `./scripts/build_and_test.sh`. The only issue is with the current version zerocopy, which is addressed by #58.

## Disclaimer

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the
terms of your choice.

### CODE REVIEW

File: src/public/log/schema.rs [hunk 1]
```diff
@@ -1060,7 +1063,12 @@ mod test {
         policy_ids.insert(PolicyId::from_str("policy2").unwrap());
 
         let errors = (0..num_of_error)
-            .map(|i| ("error".to_string() + i.to_string().as_str()))
+            .map(|i| AuthorizationError::PolicyEvaluationError {
+                id: PolicyID::from_string(format!("policy{i}")),
+                error: EvaluationError::from(ExtensionFunctionLookupError::FuncDoesNotExist {
+                    name: Name::parse_unqualified_name("foo").unwrap(),
+                }),
+            })
```
Comment (adamrothman) [inline src/public/log/schema.rs:1071]: It's no longer possible to create an `AuthorizationError` from a string so I just created some nonsense errors to appease the tests (which don't appear to care what's in the errors).

File: tests/lib.rs [hunk 2]
```diff
@@ -403,6 +405,7 @@ mod test {
     }
 
     #[tokio::test]
+    #[should_panic]
```
Comment (adamrothman) [inline tests/lib.rs:408]: I wasn't sure what to do about this test so I marked it as `#[should_panic]`. It appears to test something that is no longer valid with cedar-policy 3.0.0, but I didn't feel comfortable removing it altogether without confirmation first.

**Comment by camarcio** [comment]
Thanks @adamrothman for your contribution. We will take a look at your PR soon.

**Comment by adamrothman** [comment]
Rebased to bring in https://github.com/cedar-policy/cedar-local-agent/pull/58 to appease `cargo audit`.

File: tests/lib.rs (hunk 2 above)
Comment (camarcio) [reply tests/lib.rs:408]: Let's keep the test for now. Can you please rename it to `authorize_with_duplicated_input_entities_should_panic`?

Comment (adamrothman) [reply tests/lib.rs:408]: Done!

File: src/public/log/schema.rs (hunk 1 above)
Comment (camarcio) [reply src/public/log/schema.rs:1071]: That's ok. I will create an issue to check the contents of these errors in one of the test scenarios.

**Comment by camarcio** [comment]
Can you please apply the changes below to deny.toml?
```
diff --git a/deny.toml b/deny.toml
index 59e7f4f..712db0e 100644
--- a/deny.toml
+++ b/deny.toml
@@ -18,6 +18,7 @@ skip = [
     { name = "darling_macro", version = "=0.14.4" }, # old dependency from derive_builder
     { name = "darling_core", version = "=0.14.4" }, # old dependency from derive_builder
     { name = "bitflags", version = "=1.3.2" }, # old transitive dependency from cedar_policy_core
+    { name = "itertools", version = "=0.10.5" }, # old transitive dependency from lalrpop and criterion
     { name = "regex-syntax", version = "<=0.7.5" }, # old transitive dependency from cedar_policy_core
     { name = "windows_aarch64_gnullvm", version = "=0.48.5" }, # old dependency from chrono,lalrpop
     { name = "windows_aarch64_msvc", version = "=0.48.5" }, # old dependency from chrono,lalrpop
```

This is for fixing `cargo deny check`

**Review by camarcio — APPROVED** [review]
Looks good to me. Thanks for your contribution!

**Pull request merged**